### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by tandcode**
Migrated to Gradle, added tests, couple small fixes.

### CODE REVIEW

File: src/test/java/com/github/tandcode/task_poly_1/TestVirtualPrinter.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.github.tandcode.task_poly_1;
+
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class TestVirtualPrinter {
+
+	@Test 
+	public void testPrinter() {
+		Printer printer = new VirtualPrinter();
```
Comment (reykjavik11) [inline src/test/java/com/github/tandcode/task_poly_1/TestVirtualPrinter.java:10]: Better to use anonymous class instead VirtualPrinter class:
```
Printer printer = new Printer() {
			@Override
			public String print() {
				return super.print();
			}
		};
```

VirtualPrinter should be deleted.
TestCasting  test may be changed to:
```
public class TestCasting {

	@Test(expected = ClassCastException.class)
	public void testClassCastException() {
		Printer inkPrinter = new InkPrinter();
		Printer virtualPrinter = **(ThreeDPrinter)** inkPrinter;
	}
}
```

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+repositories {
+    // Use jcenter for resolving dependencies.
+    // You can declare any Maven/Ivy/file repository here.
+    jcenter()
+}
+
+dependencies {
+    // This dependency is used by the application.
+    implementation 'com.google.guava:guava:28.2-jre'
```
Comment (reykjavik11) [inline build.gradle:25]: This dependency looks like redundant:
// This dependency is used by the application.
    implementation 'com.google.guava:guava:28.2-jre'

File: .project [hunk 3]
```diff
@@ -1,17 +0,0 @@
...
-		<buildCommand>
-			<name>org.eclipse.jdt.core.javabuilder</name>
-			<arguments>
-			</arguments>
-		</buildCommand>
-	</buildSpec>
-	<natures>
-		<nature>org.eclipse.jdt.core.javanature</nature>
-	</natures>
-</projectDescription>
```
[the remark this answers is not in the document]
Comment (tandcode) [reply .project:17]: It was just old version in master branch, dev olready fix it.

File: src/main/java/com/github/tandcode/task_poly_1/VirtualPrinter.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.github.tandcode.task_poly_1;
+
+public class VirtualPrinter extends Printer {
+
+	public VirtualPrinter() {
+		super();
```
[the remark this answers is not in the document]
Comment (tandcode) [reply src/main/java/com/github/tandcode/task_poly_1/VirtualPrinter.java:6]: I find an interesting thing. 
```
public Printer() {
		System.out.println("Printer constructor");
	}
```

```
public InkPrinter() {
		System.out.println("InkPrinter printer constractor");
	}
```
```
@Test 
	public void testPrinter() {
		Printer inkPrinter = new InkPrinter();
	}
```
It outputs "Printer constructor" and "InkPrinter printer constractor" both, so why do we need method super() if jvm calls parent constructor in any case?

File: src/test/java/com/github/tandcode/task_poly_1/TestVirtualPrinter.java (hunk 1 above)
Comment (tandcode) [reply src/test/java/com/github/tandcode/task_poly_1/TestVirtualPrinter.java:10]: Got it.

File: build.gradle (hunk 2 above)
Comment (tandcode) [reply build.gradle:25]: Ok)

**Pull request merged**